fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/PartitionColPredicateEvaluator.java

```diff
@@ -267,14 +274,23 @@ public BitSet visitInPredicate(InPredicateOperator predicate, Void context) {
         public BitSet visitIsNullPredicate(IsNullPredicateOperator predicate, Void context) {
             predicate.getChild(0).accept(this, null);
             PartitionKey conditionKey = new PartitionKey();
-            Type columnType = Type.fromPrimitiveType(partitionColumn.getPrimitiveType());
+            Type columnType = predicate.getChild(0).getType();
             try {
-                conditionKey.pushColumn(LiteralExpr.createInfinity(columnType, false), partitionColumn.getPrimitiveType());
+                conditionKey.pushColumn(LiteralExpr.createInfinity(columnType, false), columnType.getPrimitiveType());
             } catch (AnalysisException e) {
                 return createAllTrueBitSet();
             }
             Range<PartitionKey> predicateRange = Range.closed(conditionKey, conditionKey);
-            return evaluateRangeHitSet(predicate.getChild(0), predicateRange);
+            BitSet res = evaluateRangeHitSet(predicate.getChild(0), predicateRange);
+            res.or(evaluateNullRangeHitSet(predicate.getChild(0), columnType));
+            return res;
+        }
+
+        private BitSet evaluateNullRangeHitSet(ScalarOperator scalarOperator, Type columnType) {
+            PartitionKey nullKey = new PartitionKey();
+            nullKey.pushColumn(NullLiteral.create(columnType), columnType.getPrimitiveType());
+            Range<PartitionKey> nullRange = Range.closed(nullKey, nullKey);
+            return evaluateRangeHitSet(scalarOperator, nullRange);
```

Comment:
THis will make nullRange not connected to any range, so it may return all candidate partitions. Null processing is buggy. How about return all partition directly?

Reply:
We just support col is null or col <=> null.

Reply:
I think do union with MinRnage is enough？

Reply:
Could you test a case with select * from tbl where parition_key is null when there exists null values in the first parition and normal values in other paritions? The expect prune result should be one partition.

Reply:
as you wish